### PR TITLE
feat: Add SQL Server support

### PR DESCRIPTION
**Pull request by dbuduev**
Signed-off-by: Dennis Buduev <dennis@cerbos.dev>

#### Description

Adds SQL Server as a cerbos policies storage.

#### TODO

- [ ] Manual testing
- [ ] E2E test
- [x] Documentation
 
#### Checklist 

- [x] The PR title has the correct prefix 
- [ ] PR is linked to the corresponding issue
- [x] All commits are signed-off (`git commit -s ...`) to provide the [DCO](https://developercertificate.org/)

### CODE REVIEW

File: internal/storage/db/internal/db.go [hunk 1]
```diff
@@ -263,7 +264,7 @@ func (s *dbStorage) GetCompilationUnits(ctx context.Context, ids ...namer.Module
 		Where(
 			goqu.And(
 				goqu.C(PolicyDepTblPolicyIDCol).Table("pd").In(ids),
-				goqu.C(PolicyTblDisabledCol).Table("p").Eq(false),
+				goqu.C(PolicyTblDisabledCol).Table("p").Eq(DialectFalseValue(s.db.Dialect())),
```
Comment (charithe) [inline internal/storage/db/internal/db.go:267]: Any idea why the dialect doesn't handle this conversion automatically? It seems to be configured correctly: https://github.com/doug-martin/goqu/blob/31d438d38230118d377f11cfaad9627aa3a23864/dialect/sqlserver/sqlserver.go#L33

**Review by charithe — COMMENTED** [review]
Looks OK to me although I am a bit concerned about why the dialect doesn't handle the boolean conversion correctly. 

The driver needs to be imported in the `server` package to register it (https://github.com/cerbos/cerbos/blob/9411f6289e0f6a7d3846e1033627fab6fc876e1f/internal/server/server.go#L63-L79)

File: internal/storage/db/internal/db.go (hunk 1 above)
Comment (dbuduev) [reply internal/storage/db/internal/db.go]: Not sure why it wasn't being picked up. It works if we replace `false` with `goqu.V(false)`.

**Comment by dbuduev** [comment]
I've refactored the implementations, updated docs and manually tested, but probably I haven't done enough of end-2-end testing.

**Review by charithe — APPROVED** [review]
LGTM. I like the refactoring. It's really clean 👍🏽 

Not sure whether https://github.com/microsoft/mssql-docker/tree/master/linux/sample-helm-chart works. If it does, hopefully the E2E tests won't be too hard. That could be a separate PR or part of this as well.

**Pull request merged**